### ISSUE TITLE
Support for attachments via IMessageChannel

### ISSUE THREAD

**Issue by pszmyd** [OrchardCMS/Orchard#6196]
We should at least support sending attachments via the default `ISmtpChannel`. This would be a non-breaking change, hence good for 1.9.x

**Comment by Ermesx** [comment]
I can surprise you but Orchard has that feature :) 
It's not explicit but you can do it because @sfmskywalker add this functionality in this commit: 81fb0d7

Look here in the implementation of `SmtpMessageChannel.cs`
``` c#
// Part of SmtpMessageChannel.cs
if (parameters.ContainsKey("Message")) {
// A full message object is provided by the sender.

    var oldMessage = mailMessage;
    mailMessage = (MailMessage)parameters["Message"];

    if (String.IsNullOrWhiteSpace(mailMessage.Subject))


    if (String.IsNullOrWhiteSpace(mailMessage.Body)) {
        mailMessage.Body = oldMessage.Body;
        mailMessage.IsBodyHtml = oldMessage.IsBodyHtml;
    }
}
```
You can pass whole message object with attachments.
I said about it in #6082 but I think that need small refactoring.

**Comment by pszmyd** [comment]
Thanks, I'm aware of this:) 

I was just thinking about something more generic, SMTP-agnostic. Attachments do not only apply to e-mail messages. Think of different channels (MMS, VMS, Slack, whatever etc.).

**Comment by pszmyd** [comment]
Anyways, I got the implementation based on dictionary parameters ready. Will push it so you can check:)

**Comment by HermesSbicego-Laser** [comment]
Hi guys, any news on this?

**Comment by sfmskywalker** [comment]
I think we're still waiting for a PR.

**Comment by HermesSbicego-Laser** [comment]
I have something implemented in 1.8.2; if I remeber well, attachments dictionary key has an ienumerable<string> value, representing an uri for each attachment. If compliant with your needs too I can send a PR tomorrow.

**Comment by sebastienros** [comment]
Feel free to create a PR so we can take a look at the code.
